lib/ex_aws/sns/request.ex

```diff
@@ -0,0 +1,23 @@
+defmodule ExAws.SNS.Request do
+  @moduledoc false
+  # SNS specific request logic.
+
+  @empty_body_hash AWSAuth.Utils.hash_sha256("")
+
+  def request(client, action, params) do
+    query = params
+    |> Map.put("Action", Mix.Utils.camelize(Atom.to_string(action)))
+    |> URI.encode_query
```

Comment:
OK publishing works now. In order to make this happen this branch of ExAws now depends on benwilson512/aws_auth instead of the published aws_auth package.

Before merging this branch into master I'd like to probably move auth in house so that there isn't a git based dependency on master. However, this at least lets us use this branch for our own purposes in the mean time.